python/python/lance/dataset.py

```diff
@@ -703,6 +703,18 @@ def _commit(
         """
         if isinstance(base_uri, Path):
             base_uri = str(base_uri)
+            
+        _Dataset.commit(base_uri, operation, read_version, options)
+        return LanceDataset(base_uri)
+
+
+class LanceOperation:
```

Comment:
I'm using data classes now (very cool feature I wasn't aware of).  I wanted these to be grouped together somehow so I namespaced them inside `LanceOperation`.  Let me know what you think.

Reply:
I see. So it would be `lance.LanceOperation.Append(...)`.

More canonical might be to just make it a submodule, so it's `lance.operations.Append(...)`. I'm ambivalent on it though, both seem fine to me.

Reply:
Yes.  I only avoided submodule because that would mean an additional directory and the resulting code wouldn't be very self-contained (operations don't make a ton of sense without any context).

I agree though that `lance.LanceOperation.append` is not very pythonic :(

If anyone has a strong feeling I will change

At the moment this feature is at least semi-internal (right?) so that helps I suppose